amaranth/back/rtlil.py

```diff
@@ -372,6 +373,7 @@ def emit_signal_wires(self):
             # the design is flattened) will do that anyway, so it doesn't matter.
             attrs = self.value_attrs.setdefault(value, {})
             attrs.update(signal.attrs)
+            self.value_src_loc[value] = signal.src_loc
```

Comment:
Should we not do this for other values too? Most of them have a source location (all except `Const`. And I think we should just track them for `Const` too, no reason not to)

Reply:
we're talking NIR values here, not AST values; they don't have locations

Reply:
more specifically, this change is to make sure we actually emit locations for signal wires, in the case when they also become port wires; the thing we *don't* have currently covered is anonymous ports (since they don't really have 1-1 correspondence to signals); this could also be improved, but that's a more complex issue